hypershift-operator/main.go

```diff
@@ -375,6 +376,12 @@ func filterMutableHostedClusterSpecFields(spec *hyperv1.HostedClusterSpec) {
 }
 
 func validateHostedClusterUpdate(new *hyperv1.HostedCluster, old *hyperv1.HostedCluster) admission.Response {
+	if old.Spec.ClusterID != "" && new.Spec.ClusterID != old.Spec.ClusterID {
+		return admission.Denied("spec.clusterID is immutable once set")
+	}
+	if old.Spec.InfraID != "" && new.Spec.InfraID != old.Spec.InfraID {
+		return admission.Denied("spec.InfraID is immutable once set")
+	}
```

Comment:
>return admission.Denied("attempted change to immutable field(s)")

It'd be good for UX to add the ability to return a more meaningful message granular per field, we can leverage "k8s.io/apimachinery/pkg/util/validation/field"  for that e.g field.Invalid(field.NewPath("spec", "selector")

Reply:
I agree.  Completely open to UX improvements here.  I was just looking to get the base validation in place.